File: y2021/python/day_2.py

with open('../inputs/day_2') as f:
    lines = f.readlines()

depth = 0
horizontal = 0

for line in lines:
    direction, magnitude = line.split()
    magnitude = int(magnitude)
    if direction == 'forward':
        horizontal += magnitude
    if direction == 'up':
        depth -= magnitude
    if direction == 'down':
        depth += magnitude

print(depth, horizontal)
print(depth*horizontal)

File: y2021/cpp/day2.cpp
#include <iostream>
#include <fstream>
using namespace std;


int main()
{
    ifstream inputFile ("../inputs/day_2");
    int depth = 0;
    int horizontal = 0;
    string direction;
    string magnitude;
    while (inputFile >> direction >> magnitude){
        if (direction == "forward") {
            horizontal += stoi(magnitude);
        }
        if (direction == "up"){
            depth -= stoi(magnitude);
        }
        if (direction == "down"){
            depth += stoi(magnitude);
        }
    }
    cout << depth*horizontal << endl;
}